### PR TITLE
Added some dominaria cards

### PR DESCRIPTION
**Pull request by use**
Added cards:

- Ghitu Lavarunner
- Jaya Ballard
- Wizard's Lightning

Reworked Daretti, Scrap Savant's draw/discard into a generic effect that can be reused, and Jaya uses this new effect

### CODE REVIEW

File: Mage.Sets/src/mage/cards/j/JayaBallard.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        super(card);
+    }
+
+    @Override
+    public JayaBallard copy() {
+        return new JayaBallard(this);
+    }
+}
+
+class JayaBallardEmblem extends Emblem {
```
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/j/JayaBallard.java]: Emblems should be located in mage.game.command.emblems.
I guess there was some handle image background for this.

File: Mage.Sets/src/mage/cards/j/JayaBallard.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+    public JayaBallardEmblem() {
+        this.setName("Emblem Jaya");
+        this.getAbilities().add(new SimpleStaticAbility(Zone.COMMAND, new JayaBallardEmblemContinuousEffect()));
+    }
+}
+
+class JayaBallardEmblemContinuousEffect extends ContinuousEffectImpl {
+
+    private static final FilterCard filter = new FilterCard("Instant or sorcery spell");
```
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/j/JayaBallard.java]: You can use a filter from StaticFilters.

File: Mage.Sets/src/mage/cards/j/JayaBallard.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        for (Card card : player.getGraveyard().getCards(filter, game)) {              
+            ContinuousEffect effect = new JayaBallardEmblemCastFromGraveyardEffect();
```
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/j/JayaBallard.java]: This enters the same continuous effects over an over again for each applyEffects call.
I guess you need only two continous effects that do the same for all cards in the controllers graveyard.

File: Mage.Sets/src/mage/cards/g/GhituLavarunner.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(2);
+
+        // As long as there are two or more instant and/or sorcery cards in your graveyard, Ghitu Lavarunner gets +1/+0 and has haste.
+        Ability ability = new SimpleStaticAbility(
+            Zone.BATTLEFIELD,
+            new ConditionalContinuousEffect(
+                new BoostSourceEffect(1, 0, Duration.WhileOnBattlefield),
+                SpellMasteryCondition.instance,
```
Comment (ingmargoudt) [inline Mage.Sets/src/mage/cards/g/GhituLavarunner.java]: Not sure if this is true Spell Mastery...

File: Mage.Sets/src/mage/cards/j/JayaBallard.java [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+    @Override
+    public JayaBallardEmblemCastFromGraveyardEffect copy() {
+        return new JayaBallardEmblemCastFromGraveyardEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        if (objectId.equals(getTargetPointer().getFirst(game, source)) &&
+            affectedControllerId.equals(source.getControllerId())
+        ) {
```
Comment (ingmargoudt) [inline Mage.Sets/src/mage/cards/j/JayaBallard.java]: indentation.

File: Mage.Sets/src/mage/cards/j/JayaBallard.java (hunk 3 above)
Comment (use) [reply Mage.Sets/src/mage/cards/j/JayaBallard.java]: Thank you, I will have to try a different route with this one. I re-implemented the emblem as a simple static ability, but I don't know how to tie the replacement effect to that ability. My original code was based on kess, dissident mage, which has the same bug.

File: Mage.Sets/src/mage/cards/g/GhituLavarunner.java (hunk 4 above)
Comment (use) [reply Mage.Sets/src/mage/cards/g/GhituLavarunner.java]: Thanks for the feedback, I believe spell master has no rules meaning but I understand your point, and I will submit a new PR.

**Pull request closed without merging**